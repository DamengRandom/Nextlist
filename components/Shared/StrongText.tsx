import { Text, TextProps } from '@chakra-ui/react';

const StrongText = ({ children, ...props }: TextProps) => (
  <Text as="span" fontWeight="bold" display="inline" {...props}>
    {children}
  </Text>
);

export default StrongText;
