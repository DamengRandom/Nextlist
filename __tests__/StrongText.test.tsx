import { render, screen } from '@testing-library/react';
import StrongText from '../components/Shared/StrongText';

describe('StrongText', () => {
  const testText = 'Test Text';

  it('renders the children text', () => {
    render(<StrongText>{testText}</StrongText>);
    expect(screen.getByText(testText)).toBeInTheDocument();
  });

  it('renders as a span element', () => {
    render(<StrongText>{testText}</StrongText>);
    expect(screen.getByText(testText).tagName).toBe('SPAN');
  });

  it('has bold font weight', () => {
    render(<StrongText>{testText}</StrongText>);
    const element = screen.getByText(testText);
    expect(element).toHaveStyle('font-weight: bold');
  });

  it('has inline display', () => {
    render(<StrongText>{testText}</StrongText>);
    const element = screen.getByText(testText);
    expect(element).toHaveStyle('display: inline');
  });

  it('passes through additional props', () => {
    const testId = 'strong-text';
    const color = 'red.500';
    render(
      <StrongText data-testid={testId} color={color}>
        {testText}
      </StrongText>
    );
    
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent(testText);
    // For Chakra UI specific props, you might need to check the rendered class or style
    // This depends on how Chakra processes the props
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<StrongText>{testText}</StrongText>);
    expect(asFragment()).toMatchSnapshot();
  });
});