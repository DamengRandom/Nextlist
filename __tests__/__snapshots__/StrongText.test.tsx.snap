// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StrongText matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="chakra-text css-6u8cgc"
  >
    Test Text
  </span>
</DocumentFragment>
`;
